packages/api/lambdas/bulk-operation.js

```diff
@@ -20,10 +22,18 @@ async function applyWorkflowToGranules({
   meta,
   queueUrl,
   granuleModel = new GranuleModel(),
+  granulePgModel = new GranulePgModel(),
+  granuleTranslateMethod = translatePostgresGranuleToApiGranule,
+  knex,
 }) {
   const applyWorkflowRequests = granuleIds.map(async (granuleId) => {
     try {
-      const granule = await granuleModel.get({ granuleId });
+      // TODO - this still has legacy dynamo behavior where it's not selecting
+      // based on collection/granuleId
```

Comment:
I think this is where we decided to leave it for now, at least in the endpoint logic. Do you think this should be adapted to use granule ID and collection ID in this PR? I'm thinking no only because the approach needs to be worked out and will probably apply to the endpoint and everything that comes after.

Reply:
So - I think what came out of our conversation, and my opinion in general is becoming that .get calls should *generally* be queries for unique keys, and perhaps we made a mistake in not constraining it. 

In the specific case of granules, given the theoretical case that there can be records with duplicate granuleIds in the table, it feels consistent to find a sane way to cover the edge case that a `.first()` creates when doing a `.get` with only a `granuleId`, in lieu of an implementation that sorts the Phase 3 case. 

I think you're right that we shouldn't implement a granuleId/collectionId solution here.

Reply:
>it feels consistent to find a sane way to cover the edge case that a .first() creates when doing a .get with only a granuleId

Are you saying that we should revise the current `get` behavior here? Something like the `.search()` implementation with error handling around multiple results we did elsewhere? `get` _sort of_ does that where it'll just return one of the granules matching the ID but unless we sort there's no guarantee which granule it would return and if there are many we wouldn't know because there's no check/error handling.

I guess I haven't really looked at `applyWorkflowToGranules` to figure out where this might matter. I'm viewing it as downstream from the PUT endpoint that operates on only a `granule_id`, which makes me think it fine (for now) to match that behavior here. But that could be way off base.

Reply:
> Are you saying that we should revise the current get behavior here? Something like the .search() implementation with error handling around multiple results we did elsewhere?  

Yeah, I think we should use `getUniquePgGranuleByGranuleId` - while it's blocked for now by parity with Dynamo and Dyanmo rejecting duplicates, I think we should be defensive here until a more robust lookup (granuleId/collection) lookup is implemented, particularly given .get's .first() behavior.

Reply:
That makes sense. Right now if we try to get a granule and there are duplicates, we should throw an error because it would be unclear which was intended.

https://github.com/nasa/cumulus/pull/2415/commits/7e56e8d4cc8793a82a1f35b8fe282000b071c9b5